core/frontend/src/IModelConnection.ts

```diff
@@ -550,21 +579,33 @@ export abstract class IModelConnection extends IModel {
     if (this.noGcsDefined)
       return cartographic.map((p) => this.cartographicToSpatialFromEcef(p));
 
+    const geoCoords = cartographic.map((p) => Point3d.create(p.longitudeDegrees, p.latitudeDegrees, p.height));
+    return this.toSpatialFromGcs(geoCoords);
+  }
+
+  /** Convert geographic coordinates into points in this iModel's spatial coordinate system using a [[GeoConverter]] or the iModel's [EcefLocation]($common).
+   * @param geoCoords Coordinates to be converted are in the coordinate system described by the `datumOrGCRS` parameter.  Defaults iModel's spatial coordinate system otherwise.
+   * @param datumOrGCRS Datum name or Geographic CRS object definition to use for the conversion.
+   * @returns The `geographics` coordinates converted to spatial coordinates, of the same length and order as `geographics`.
+   * @throws IModelError if [[isGeoLocated]] is false or any point could not be converted.
+   * @see [[cartographicFromSpatial]] to perform the inverse conversion.
```

Comment:
That doesn't perform the inverse - it produces `Cartographic[]` not `XYZProps[]`.